src/std/misc/string.ss

```diff
@@ -215,3 +218,29 @@
 	(string-set! str i (random-word-char)))
       str)
     ""))
+
+;; str converts all of its arguments into a single string.
+;; When called without an argument an empty string is returned.
+;;
+;; Examples:
+;;  (str 2.0)               => "2.0"
+;;  (str "hello" ", world") => "hello, world"
+(def* str
+  ((v) (if (string? v) v
+           (format ((make-str-fmt) v) v)))
+  (xs (call-with-output-string
+       (lambda (port)
+         (let (fmt-proc (make-str-fmt))
+           (let loop ((rest xs))
+             (when (pair? rest)
+               (let (v (car rest))
+                 (if (string? v)
+                   (write-string v port)
+                   (fprintf port (fmt-proc v) v)))
+               (loop (cdr rest)))))))))
+
+(def (make-str-fmt)
+  (let (obj-pr? (cut method-ref <> ':pr))
+    (lambda (v) (cond ((? (and number? inexact?) v)  "~f")
+                 ((? (or list? hash-table? vector? ##values? obj-pr?) v) "~r")
+                 (else "~a")))))
```

Comment:
you know, this can be written more concisely with match:
```
(match <>
  ((? (and number? inexact?)) "~f")
  ((? (or list? hash-table? vector? ##values? obj-pr?)) "~r")
  (else "~a"))
```
```

Reply:
Also, this should be a top-level procedure.

Reply:
Call it `str-format`, it takes a value and returns its format string.
It can be defined as
```
(def (str-format v)
  (def (obj-pr? v) ...)
  (match v
    ...))
```